doc/nrf/protocols/matter/getting_started/custom_clusters.rst

```diff
@@ -0,0 +1,408 @@
+   To see the full list of available elements for each part of the XML file, refer to the Matter Specification.
+
+You can use the following template for the :file:`MyCluster.xml` file:
+
+.. code-block:: xml
+
+   <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+   <configurator>
+     <enum name="MyNewEnum" type="int8u">
+       <cluster code="0xFFF1FC01" />
+       <item name="EnumValue1" value="0" />
+       <item name="EnumValue2" value="1" />
+     </enum>
+     <struct name="MyStruct" isFabricScoped="true">
+       <cluster code="0xFFF1FC01"/>
+       <item fieldId="1" name="Data" type="octet_string" length="128" isFabricSensitive="true"/>
+     </struct>
+     <cluster>
+       <domain>General</domain>
+       <name>MyNewCluster</name>
+       <code>0xFFF1FC01</code>
+       <define>MY_NEW_CLUSTER</define>
+       <description>The MyNewCluster cluster showcases a cluster manufacturer extensions</description>
+       <attribute side="server" code="0x00" define="FLIP_FLOP" type="boolean" writable="true" default="false" optional="false">MyAttribute</attribute>
+       <command source="client" code="0x02" name="MyCommand" optional="false">
+         <description>Command that takes two uint8 arguments and returns their sum.</description>
+         <arg name="arg1" type="int8u"/>
+         <arg name="arg2" type="int8u"/>
+       </command>
+       <event source="server" code="0x01" name="MyEvent" optional="false">
+         <description>Event that is generated by the server.</description>
+         <arg name="arg1" type="int8u"/>
+       </event>
+     </cluster>
+     <deviceType>
+       <name>my-new-device</name>
+       <domain>CHIP</domain>
+       <typeName>My new device</typeName>
+       <profileId editable="false">0x0FFF</profileId>
+       <deviceId editable="false">0x001</deviceId>
+       <class>Simple</class>
+       <scope>Endpoint</scope>
+       <clusters lockOthers="true">
+         <include cluster="MyNewCluster" client="true" server="true" clientLocked="false" serverLocked="false"/>
+       </clusters>
+     </deviceType>
+   </configurator>
+
+
+For the guide purposes, save this file as :file:`MyCluster.xml` in the sample directory.
+
+.. rst-class:: numbered-step
+
+Add the cluster description file to the data model definition file and run the ZAP tool
+***************************************************************************************
+
+The data model definition file contains all the cluster XML locations and manufacturers list.
+To work with the new custom cluster, you need to append it to the list in the existing data model definition file.
+
+You can use the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool_gui` to add the cluster and run the ZAP tool, or :ref:`ug_matter_gs_tools_matter_west_commands_append` to add the cluster only without starting the ZAP tool.
+This guide has focus on the :ref:`ug_matter_gs_tools_matter_west_commands_zap_tool_gui`.
+
+1. Run the following command:
```

Comment:
It is worth mentioning that this command must be called from the sample's directory. Not sure how to write it here properly.

Reply:
will add a note